src/react/hooks/internal/wrapHook.ts

```diff
@@ -81,15 +88,34 @@ export function wrapHook<Hook extends (...args: any[]) => any>(
   useHook: Hook,
   clientOrObsQuery: ObservableQuery<any> | ApolloClient
 ): Hook {
-  const queryManager = (
-    clientOrObsQuery as unknown as {
-      // both `ApolloClient` and `ObservableQuery` have a `queryManager` property
-      // but they're both `private`, so we have to cast around for a bit here.
-      queryManager: QueryManagerWithWrappers;
+  // Priority-wise, the later entries in this array wrap
+  // previous entries and could prevent them (and in the end,
+  // even the original hook) from running
+  const wrapperSources = [
+    (
+      clientOrObsQuery as unknown as {
+        // both `ApolloClient` and `ObservableQuery` have a `queryManager` property
+        // but they're both `private`, so we have to cast around for a bit here.
+        queryManager: QueryManagerWithWrappers;
+      }
+    )["queryManager"],
+    // if we are a hook (not `preloadQuery`), we are guaranteed to be inside of
+    // a React render and can use context
+    hookName.startsWith("use") ?
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      React.useContext(getApolloContext())
+    : undefined,
+  ];
+
+  let wrapped = useHook;
+  for (const source of wrapperSources) {
+    const wrapper = source?.[wrapperSymbol]?.[hookName] as
+      | undefined
+      | ((wrap: Hook) => Hook);
+    if (wrapper) {
+      wrapped = wrapper(wrapped);
```

Comment:
```suggestion
    const wrapper = source?.[wrapperSymbol]?.[hookName];
    if (wrapper) {
      wrapped = wrapper(wrapped) as Hook;
```

It looks like you could simplify this and reduce a bit of noise by casting the `wrapper` call instead.

Reply:
Seems to work :+1: